internal/aks/create.go

```diff
@@ -151,7 +163,7 @@ func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient
 			AuthorizedIPRanges: spec.AuthorizedIPRanges,
 		}
 	}
-	if spec.PrivateCluster != nil && *spec.PrivateCluster {
+	if to.Bool(spec.PrivateCluster) {
```

Comment:
If PrivateCluster is nil, will this cause it to be overwritten with false in upstream if it is true?

Reply:
Good catch!

Reply:
Yeap, that's right

Reply:
Is there any logic for disabling this if privateCluster is set to false in the spec but it is currently enabled upstream?

Reply:
yes, it is

Reply:
this is simple change which is simplifying this `if` condition, but `spec.PrivateCluster != nil && *spec.PrivateCluster` is  equivalent of `to.Bool(spec.PrivateCluster)`

Reply:
But what if spec.PrivateCluster != niil && !*spec.PrivateCluster? How would upstream PrivateCluster be disabled?

Reply:
After enabling `PrivateCluster` it is not possible to switch it off, this is limitation of Azure API. We don't have to worry about it.